### ISSUE TITLE
adaptor train: config-driven LoRA training pipeline wrapping mlx_lm.lora

### ISSUE THREAD

**Issue by falese** [falese/coder#8]
## Overview
Wrap `mlx_lm lora` with a reproducible, config-driven training workflow. Output is a versioned `.safetensors` file and an updated `manifest.json`.

## Command

```
coder adaptor train --config <path/to/train-config.toml>
```

## Train config schema (`train-config.toml`)

User-facing config is TOML. The CLI translates it into a temporary YAML file passed to `mlx_lm lora -c <yaml>` (LoRA params like rank and target_modules cannot be passed as CLI flags — they require the YAML config format).

```toml
[model]
path = "~/.coder/models/Qwen2.5-Coder-7B"

[lora]
rank = 8
target_modules = ["q_proj", "v_proj"]
iters = 1000
batch_size = 4
learning_rate = 1e-4

[data]
dir = "./data"            # must contain train.jsonl and valid.jsonl
                          # coder data split outputs valid.jsonl directly

[output]
adaptor_dir = "./weights"
manifest = "./manifest.json"
log_file = "./training.log"
```

## mlx_lm.lora invocation

```
python3 -m mlx_lm lora \
  --model <path> \
  --train \
  --data <dir> \
  --iters <n> \
  --batch-size <n> \
  --learning-rate <lr> \
  --adapter-path <weights-dir> \
  --mask-prompt \
  --grad-checkpoint \
  -c <generated-yaml>      # contains rank, target_modules, lora_layers
```

Checkpoint resumption: if `weights/adaptor.safetensors` exists, automatically add `--resume-adapter-file weights/adaptor.safetensors`. No user flag needed.

## Generated YAML (internal, not user-facing)

```yaml
lora_layers: 16
lora_parameters:
  rank: 8
  alpha: 16
  dropout: 0.0
  scale: 10.0
```

`target_modules` maps to `lora_layers` count — `q_proj`/`v_proj` on 16 layers covers the standard LoRA configuration for 7B models.

## Acceptance criteria

- [ ] Validates TOML config before starting; clear errors for missing/invalid fields (Zod)
- [ ] Generates correct YAML and CLI args from TOML config
- [ ] Spawns `python3 -m mlx_lm lora` with `--mask-prompt` and `--grad-checkpoint`
- [ ] Streams training progress (step, loss) to stdout in real time
- [ ] Writes per-step loss to `log_file` as structured JSON lines
- [ ] Automatic checkpoint resumption when `weights/adaptor.safetensors` exists
- [ ] On completion, updates `manifest.json` with bumped patch version and final eval loss
- [ ] Unit tests: config validation, YAML generation, arg construction (mock subprocess)
- [ ] Integration test: dry-run mode exits cleanly without spawning Python
- [ ] `tsc --noEmit` clean, ESLint clean

## Notes
- Training loss curve is a required observability metric per the base spec
- `--grad-checkpoint` reduces memory use — important for staying within 18 GB
- `--mask-prompt` ensures the model learns to generate completions, not re-predict prompts
- `coder data split` outputs `valid.jsonl` (fixed in #7) — matches `mlx_lm lora --data` expectations directly
